Updated packages.md and added --experimental-detect-module flag (#51089)
nodejs/node: doc/api/packages.md

* Strings passed in as an argument to `--eval`, or piped to `node` via `STDIN`,
  with the flag `--input-type=module`.

> * When using [--experimental-detect-module](https://nodejs.org/docs/latest/api/cli.html),

Reviewer: Like other links, it needs to be a reference you add at the end of the document.
```suggestion
* When using [`--experimental-detect-module`][]
```

Author: I didn't know that. Thanks.

> * When using [`--experimental-detect-module`][], code that contains syntax that
>   only parses successfully as [ES modules][], such as `import` or `export`

Reviewer 2: The current is kind of difficult to follow (too many subordinate clauses).

```suggestion
* When using [`--experimental-detect-module`][], code containing syntax only successfully parsed as [ES modules][], such as `import` or `export`
```

Line-length in the suggestion is probably not compliant (and the paragraph will need to be adjusted).

Author: I made few changes. Please check.

> * Code that contains syntax that only parses successfully as [ES modules][],

Reviewer: ```suggestion
* When using [`--experimental-detect-module`][], code that contains syntax that
  only parses successfully as [ES modules][],
```

Author: Thanks
